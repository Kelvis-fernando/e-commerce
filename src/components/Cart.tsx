import {
  CartContainer,
  CartContent,
  CloseCart,
  TotalCart,
} from "../styles/cart";
import { CartProps } from "../types/cart";
import { X, Trash } from "phosphor-react";
import { removeItemFromCart, store } from "../states/cartState";
import { ProductCardProps } from "../types/productCardProps";
import { useState, useEffect } from "react";

const Cart = ({ isOpen, onClose }: CartProps) => {
  const [cartItems, setCartItems] = useState(store.getState().items);

  useEffect(() => {
    const unsubscribe = store.subscribe(() => {
      setCartItems(store.getState().items);
    });
    return unsubscribe;
  }, []);

  const handleRemoveItem = (item: ProductCardProps) => {
    store.dispatch(removeItemFromCart(item.id));
  };

  if (!isOpen) return null;
  return (
    <CartContainer isOpen={isOpen}>
      <CartContent>
        <CloseCart>
          <X onClick={onClose} />
        </CloseCart>
        <h1>Carrinho</h1>
        <ul>
          {cartItems.length > 0 ? (
            cartItems.map((item: ProductCardProps, index: number) => (
              <li key={index}>
                <span>{item?.name}</span>
                <Trash size={16} onClick={() => handleRemoveItem(item)} />
              </li>
            ))
          ) : (
            <li>
              <span>nenhum item adicionado</span>
            </li>
          )}
        </ul>
        <TotalCart>
          <h1>Total:</h1>
          <p>
            R${" "}
            {cartItems
              .map((price) => price.price)
              .reduce((total, actual) => {
                return total + actual;
              }, 0)
              .toFixed(2)}
          </p>
        </TotalCart>
        <button>Finalizar compra</button>
      </CartContent>
    </CartContainer>
  );
};

export default Cart;
