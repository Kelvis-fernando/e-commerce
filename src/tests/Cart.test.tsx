import React from "react";
import { render } from "@testing-library/react";
import Cart from "../components/Cart";

describe("Cart", () => {
  test("should render and match snapshot", () => {
    const { container } = render(
      <Cart
        isOpen={false}
        onClose={function (): void {
          throw new Error("Function not implemented.");
        }}
      />
    );

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
