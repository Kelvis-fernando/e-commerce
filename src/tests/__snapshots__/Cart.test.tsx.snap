// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should render and match snapshot 1`] = `<div />`;
